src/config.cpp

```diff
@@ -736,21 +766,50 @@ CParseResult CConfig::parseFile(const char* file) {
         return result;
     }
 
-    std::string line    = "";
-    int         linenum = 1;
+    std::string rawLine    = "";
+    int         rawLineNum = 0;
+    std::string line       = "";
+    int         lineNum    = 0;
 
-    while (std::getline(iffile, line)) {
+    while (std::getline(iffile, rawLine)) {
+        line = rawLine;
+        lineNum = ++rawLineNum;
+
+        bool mergeWithNextLine = rawLine.length() > 0 && rawLine.at(rawLine.length() - 1) == '\\';
+        bool gotWholeLine = true;
+
+        while (mergeWithNextLine) {
+            const auto lastNonSpace = line.length() < 2 ? -1 : line.find_last_not_of(MULTILINE_SPACE_CHARSET, line.length() - 2);
+            line = line.substr(0, lastNonSpace + 1);
+
+            if (!std::getline(iffile, rawLine)) {
+                if (!impl->parseError.empty())
+                    impl->parseError += "\n";
+                impl->parseError += std::format("Config error in file {}: Last line ends with backslash", file);
+                result.setError(impl->parseError);
+                gotWholeLine = false;
+                break;
+            }
+
+            ++rawLineNum;
+            line += rawLine;
+            mergeWithNextLine = line.length() > 0 && line.at(line.length() - 1) == '\\';
+        }
+
+        if (!gotWholeLine)
```

Comment:
considering all of these are duplicated, can we make this into a helper function that takes a ref to iffile and returns an std::expected<std::string, int>? If it returns an int, it means fail (break and die)

Reply:
Sounds good. I actually tried to make a helper function at first, but I consolidated too much logic and accidentally changed the behavior of one of the functions. I'll take another stab at it.